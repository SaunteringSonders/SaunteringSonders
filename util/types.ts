type BannerPosition = 'top' | 'bottom' | 'text' | 'none'
type AvatarPosition = 'left' | 'right' | 'center' | 'none'

export type { BannerPosition, AvatarPosition }
