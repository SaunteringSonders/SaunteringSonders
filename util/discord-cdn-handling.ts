const DiscordCDNAttachmentRegex =
  /^https:\/\/cdn\.discordapp\.com\/attachments\/(.+)$/i

export { DiscordCDNAttachmentRegex }
