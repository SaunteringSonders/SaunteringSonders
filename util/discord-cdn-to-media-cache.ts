const DiscordCDNToMediaCache = /^https:\/\/cdn\.discordapp\.com\//i

export default DiscordCDNToMediaCache
