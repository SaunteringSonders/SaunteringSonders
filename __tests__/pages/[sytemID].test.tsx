import { render } from '@testing-library/react'
import Front from '../../pages/[systemID]'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const getFrontersAndSystem = jest.spyOn(
  require('../../util/front-utils'),
  'getFrontersAndSystem',
)

describe('fronters and system cards', () => {
  it('renders the footer only when an invalid system ID is provided', () => {
    useRouter.mockImplementation(() => ({
      query: {
        systemID: 'invalid',
      },
    }))
    getFrontersAndSystem.mockImplementation(() => {
      return {
        fronters: null,
        system: null,
      }
    })

    const { container } = render(<Front />)
    expect(container).toMatchSnapshot()
  })
})
