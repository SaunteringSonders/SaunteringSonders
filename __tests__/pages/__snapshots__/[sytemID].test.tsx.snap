// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fronters and system cards renders the footer only when an invalid system ID is provided 1`] = `
<div>
  <div
    class="flex flex-col justify-between min-h-screen"
  >
    <div
      class="container mx-auto p-2 flex flex-row flex-wrap justify-center gap-2"
    />
    <footer
      class="container mx-auto mb-2 max-w-md md:max-w-4xl p-2 bg-slate-100 dark:bg-slate-800 text-slate-900 dark:text-slate-50 text-center text-sm rounded"
    >
      Powered by 
      <span
        class="italic"
      >
        a lot
      </span>
       of 
      <a
        href="https://github.com/systemic-chaos/currently-fronting/blob/main/package.json"
      >
        open source software
      </a>
       and designed to interface with 
      <a
        href="https://pluralkit.me/"
      >
        PluralKit
      </a>
      . Developed by 
      <a
        href="https://github.com/Arcana1321"
      >
        Arcana
      </a>
       and available under the 
      <a
        href="https://github.com/systemic-chaos/currently-fronting/blob/main/LICENSE"
      >
        AGPL 3.0 license
      </a>
      . Find a bug? Follow the instructions on 
      <a
        href="https://github.com/systemic-chaos/currently-fronting"
      >
        GitHub
      </a>
       in the README.
    </footer>
  </div>
</div>
`;
